lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -555,6 +555,27 @@ struct ConstantOpConv : public OpConversionPattern<ConstantOp> {
   }
 };
 
+struct StringConstantOpConv : public OpConversionPattern<StringConstantOp> {
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(moore::StringConstantOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    const auto str = op.getValue();
+    const unsigned byteWidth = str.size() * 8;
+    Type resultType = typeConverter->convertType(op.getResult().getType());
+    resultType.dump();
+    APInt value(byteWidth, 0);
+    for (size_t i = 0; i < str.size(); ++i) {
+      const auto asciiChar = static_cast<uint8_t>(str[i]);
+      value |= APInt(byteWidth, asciiChar) << (8 * (str.size() - 1 - i));
+    }
```

Comment:
Maybe we can use `getAsInteger`? https://llvm.org/doxygen/classllvm_1_1StringRef.html#adc9c54d457a9ebc9d909c4b2234c7657

Reply:
I think `getAsInteger` converts the APInt to a string representation of the number. SystemVerilog wants you to take the actual ASCII byte values of each character and concatenate them into a large integer. So it's basically going from a string to an integer, but not by parsing the string as a number like "0x42abc", but by actually taking the ASCII values of the characters and concatenating them 😁. It's pretty wild.

Reply:
Oh, yeah. Thank you for pointing it out 👍